File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0){
            return 0;
        }

        // sort(nums.begin(),nums.end());

        // int size=1;
        // int ans=1;

        // for(int i=0;i<nums.size()-1;i++){
        //     if(nums[i+1]==nums[i]+1){
        //         size++;
        //     }
        //     else if(nums[i+1]==nums[i]){
        //         continue;
        //     }
        //     else{
        //         size=1;
        //     }
        //     ans=max(ans,size);
        // }

        // return ans;

        int size=0;
        int ans=0;
        map<int,int>mp;

        for(int n:nums){
            mp[n]++;
        }

        for(int i=0;i<nums.size();i++){
            int curr=nums[i];
            size=0;
            if(mp[curr-1]==0){
                while(mp[curr++]){
                    size++;
                }
                ans=max(ans,size);
            }
        }
        return ans;


        
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {

        Map<Integer,Integer>mp=new HashMap<>();
        int n=nums.length;

        for(int i=0;i<nums.length;i++){
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        }

        List<Integer>numsCopy=new ArrayList<>();

        for(int i=0;i<n;i++){
            numsCopy.add(nums[i]);
        }

        Collections.sort(numsCopy,(a,b)->{
            // if(mp.get(a)>mp.get(b)){
            //     return -1;
            // }
            // return 1;
            return mp.get(b)-mp.get(a);
        });

        Set<Integer>st=new LinkedHashSet<>();
        for(int i=0;i<n;i++){
            st.add(numsCopy.get(i));
        }

        Iterator<Integer>iter=st.iterator();

        int []ans=new int[k];
        int count=0;
        while(iter.hasNext()){
            if(count==k){
                break;
            }
            ans[count]=iter.next();
            count++;
        }

        return ans;  
    }
}
